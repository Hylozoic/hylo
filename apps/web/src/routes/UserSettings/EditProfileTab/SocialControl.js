import React, { Component } from 'react'
import { withTranslation } from 'react-i18next'
import PropTypes from 'prop-types'
import { Facebook, Twitter, Linkedin, Check, Link as LinkIcon, Unlink, Link2Off } from 'lucide-react'
import Button from 'components/ui/button'

const { func, string } = PropTypes

class SocialControl extends Component {
  static propTypes = {
    label: string,
    provider: string,
    value: string,
    updateSettingDirectly: func,
    handleUnlinkAccount: func
  }

  windowPrompt (network, urlPattern) {
    const promptText = this.props.t('Please enter the full url for your {{network}} page.', { network })
    const invalidUrlText = this.props.t('Invalid url. Please enter the full url for your {{network}} page.', { network })

    let url = window.prompt(promptText)

    if (url) {
      while (!url && !url.match(urlPattern)) {
        url = window.prompt(invalidUrlText)
      }
    }
    return url
  }

  handleLinkClick () {
    const { provider, updateSettingDirectly } = this.props

    switch (provider) {
      case 'twitter': {
        const twitterHandle = window.prompt(this.props.t('Please enter your twitter name.'))
        if (twitterHandle) {
          updateSettingDirectly()(twitterHandle)
        }
        break
      }
      case 'linkedin': {
        const network = 'LinkedIn'
        const urlPattern = /^(http(s)?:\/\/)?([\w]+\.)?linkedin\.com/
        const url = this.windowPrompt(network, urlPattern)
        if (url) {
          updateSettingDirectly()(url)
        }
        break
      }
      case 'facebook': {
        const network = 'Facebook'
        const urlPattern = /^(http(s)?:\/\/)?([\w]+\.)?facebook\.com/
        const url = this.windowPrompt(network, urlPattern)
        if (url) {
          updateSettingDirectly()(url)
        }
        break
      }
    }
  }

  handleUnlinkClick () {
    const { handleUnlinkAccount, updateSettingDirectly } = this.props

    handleUnlinkAccount()
    updateSettingDirectly()(null)
  }

  getSocialIcon () {
    const { provider } = this.props
    const iconProps = {
      className: 'w-5 h-5'
    }

    switch (provider) {
      case 'facebook':
        return <Facebook {...iconProps} />
      case 'twitter':
        return <Twitter {...iconProps} />
      case 'linkedin':
        return <Linkedin {...iconProps} />
      default:
        return null
    }
  }

  render () {
    const { label, value = '', t } = this.props
    const linked = !!value

    return (
      <div className='relative group bg-background/50 hover:bg-background border border-foreground/10 hover:border-foreground/20 rounded-lg p-4 transition-all duration-300'>
        <div className='flex items-center justify-between'>
          <div className='flex items-center gap-3'>
            {this.getSocialIcon()}
            <div className='flex flex-col'>
              <span className='text-sm font-medium text-foreground'>{label}</span>
              {linked && (
                <span className='text-xs text-foreground/50 truncate max-w-[200px]'>{value}</span>
              )}
            </div>
            {linked && (
              <div className='flex items-center ml-2 text-selected text-sm bg-selected/10 rounded-lg p-2'>
                <Check className='w-4 h-4 text-selected' /> {t('Linked')}
              </div>
            )}
            {!linked && (
              <div className='flex items-center ml-2 text-foreground/50 gap-1 text-sm p-2 bg-black/20 rounded-lg'>
                <Link2Off className='w-4 h-4' /> {t('Not Linked')}
              </div>
            )}
          </div>
          <Button
            variant='outline'
            size='sm'
            onClick={linked ? () => this.handleUnlinkClick() : () => this.handleLinkClick()}
            className='opacity-0 group-hover:opacity-100 transition-opacity'
          >
            {linked
              ? (
                <div className='flex items-center gap-2'>
                  <Unlink className='w-4 h-4' />
                  <span>{t('Unlink')}</span>
                </div>
                )
              : (
                <div className='flex items-center gap-2'>
                  <LinkIcon className='w-4 h-4' />
                  <span>{t('Link')}</span>
                </div>
                )}
          </Button>
        </div>
      </div>
    )
  }
}

export default withTranslation()(SocialControl)
