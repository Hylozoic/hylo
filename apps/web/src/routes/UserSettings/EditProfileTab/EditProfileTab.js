import React, { useState, useEffect } from 'react'
import { withTranslation } from 'react-i18next'
import { get } from 'lodash/fp'
import PropTypes from 'prop-types'
import { Helmet } from 'react-helmet'
import SettingsControl from 'components/SettingsControl'
import SkillsSection from 'components/SkillsSection'
import SkillsToLearnSection from 'components/SkillsToLearnSection'
import Button from 'components/ui/button'
import Icon from 'components/Icon'
import UploadAttachmentButton from 'components/UploadAttachmentButton'
import Loading from 'components/Loading'
import { useViewHeader } from 'contexts/ViewHeaderContext'
import { DEFAULT_BANNER } from 'store/models/Me'
import classes from './EditProfileTab.module.scss'
import { ensureLocationIdIfCoordinate } from 'components/LocationInput/LocationInput.store'
import SocialControl from './SocialControl'
import { bgImageStyle, cn } from 'util/index'

export const validateName = name => name && name.match(/\S/gm)

function EditProfileTab ({
  currentUser,
  updateUserSettings,
  fetchPending,
  fetchLocation,
  unlinkAccount,
  setConfirm,
  t
}) {
  const [edits, setEdits] = useState({})
  const [changed, setChanged] = useState(false)

  useEffect(() => {
    setEditState()
  }, [])

  useEffect(() => {
    if (fetchPending === false) {
      setEditState()
    }
  }, [fetchPending])

  const setEditState = () => {
    if (!currentUser) return

    const {
      name, avatarUrl, bannerUrl, tagline, bio,
      contactEmail, contactPhone, locationObject, location,
      url, facebookUrl, twitterName, linkedinUrl
    } = currentUser

    setEdits({
      name: name || '',
      avatarUrl: avatarUrl || '',
      bannerUrl: bannerUrl || DEFAULT_BANNER,
      tagline: tagline || '',
      bio: bio || '',
      contactPhone: contactPhone || '',
      contactEmail: contactEmail || '',
      location: location || '',
      locationId: get('id', locationObject) || null,
      url: url || '',
      facebookUrl,
      twitterName,
      linkedinUrl
    })
  }

  const updateSetting = (key, shouldSetChanged = true) => async event => {
    const newEdits = { ...edits }
    shouldSetChanged && setConfirm(t('You have unsaved changes, are you sure you want to leave?'))

    if (key === 'location') {
      newEdits.location = event.target.value.fullText
      newEdits.locationId = await ensureLocationIdIfCoordinate({
        fetchLocation,
        location: newEdits.location,
        locationId: event.target.value.id
      })
    } else {
      newEdits[key] = event.target.value
    }

    setEdits(newEdits)
    if (shouldSetChanged) {
      setChanged(true)
    }
  }

  const updateSettingDirectly = (key, shouldSetChanged = true) => value =>
    updateSetting(key, shouldSetChanged)({ target: { value } })

  const save = () => {
    setChanged(false)
    setConfirm(false)
    updateUserSettings(edits)
  }

  const { setHeaderDetails } = useViewHeader()
  useEffect(() => {
    setHeaderDetails({
      title: t('Edit Your Profile'),
      icon: '',
      info: '',
      search: false
    })
  }, [])

  if (fetchPending || !currentUser) return <Loading />

  const {
    name, avatarUrl, bannerUrl, tagline, bio,
    contactEmail, contactPhone, location, url,
    facebookUrl, twitterName, linkedinUrl
  } = edits

  const locationObject = currentUser.locationObject

  return (
    <div>
      <Helmet>
        <title>{t('Edit Your Profile')} | Hylo</title>
      </Helmet>
      <label className={classes.label}>{t('Banner and Avatar Images')}</label>
      <UploadAttachmentButton
        type='userBanner'
        id={currentUser.id}
        onSuccess={({ url }) => updateSettingDirectly('bannerUrl')(url)}
        className={classes.changeBanner}
      >
        <div style={bgImageStyle(bannerUrl)} className={classes.bannerImage}><Icon name='AddImage' className={classes.uploadIcon} /></div>
      </UploadAttachmentButton>
      <UploadAttachmentButton
        type='userAvatar'
        id={currentUser.id}
        onSuccess={({ url }) => updateSettingDirectly('avatarUrl')(url)}
        className={classes.changeAvatar}
      >
        <div style={bgImageStyle(avatarUrl)} className={classes.avatarImage}><Icon name='AddImage' className={classes.uploadIcon} /></div>
      </UploadAttachmentButton>
      <SettingsControl id='nameField' label={t('Name')} onChange={updateSetting('name')} value={name || ''} maxLength={60} />
      {!validateName(name) && <div className={classes.nameValidation}>{t('Name must not be blank')}</div>}
      <SettingsControl id='taglineField' label={t('Tagline')} onChange={updateSetting('tagline')} value={tagline} maxLength={60} />
      <SettingsControl id='bioField' label={t('About Me')} onChange={updateSetting('bio')} value={bio} type='textarea' />
      <SettingsControl
        id='locationField'
        label={t('Location')}
        onChange={updateSettingDirectly('location', true)}
        location={location}
        locationObject={locationObject}
        type='location'
      />
      <SettingsControl id='urlField' label={t('Website')} onChange={updateSetting('url')} value={url} />
      <SettingsControl
        label={t('My Skills & Interests')} renderControl={() =>
          <SkillsSection personId={currentUser.id} />}
      />
      <SettingsControl
        label={t('What I\'m learning')} renderControl={() =>
          <SkillsToLearnSection personId={currentUser.id} />}
      />
      <SettingsControl id='contactEmailField' label={t('Contact Email')} onChange={updateSetting('contactEmail')} value={contactEmail} />
      <SettingsControl id='contactPhoneField' label={t('Contact Phone')} onChange={updateSetting('contactPhone')} value={contactPhone} />
      <label className={classes.socialLabel}>{t('Social Accounts')}</label>
      <SocialControl
        label='Facebook'
        provider='facebook'
        value={facebookUrl}
        updateSettingDirectly={() => updateSettingDirectly('facebookUrl')}
        handleUnlinkAccount={() => unlinkAccount('facebook')}
      />
      <SocialControl
        label='Twitter'
        provider='twitter'
        value={twitterName}
        updateSettingDirectly={() => updateSettingDirectly('twitterName')}
        handleUnlinkAccount={() => unlinkAccount('twitter')}
      />
      <SocialControl
        label='LinkedIn'
        provider='linkedin'
        value={linkedinUrl}
        updateSettingDirectly={() => updateSettingDirectly('linkedinUrl')}
        handleUnlinkAccount={() => unlinkAccount('linkedin')}
      />
      <div style={{ height: '80px' }} />
      <div className={classes.saveChanges}>
        <span className={cn({ [classes.settingChanged]: changed })}>{changed ? t('Changes not saved') : t('Current settings up to date')}</span>
        <Button
          variant={changed && validateName(name) ? 'secondary' : 'primary'}
          onClick={changed && validateName(name) ? save : null}
          className={classes.saveButton}
        >
          {t('Save Changes')}
        </Button>
      </div>
    </div>
  )
}

EditProfileTab.propTypes = {
  currentUser: PropTypes.object,
  updateUserSettings: PropTypes.func,
  fetchPending: PropTypes.bool,
  fetchLocation: PropTypes.func,
  unlinkAccount: PropTypes.func,
  setConfirm: PropTypes.func,
  t: PropTypes.func
}

export default withTranslation()(EditProfileTab)
