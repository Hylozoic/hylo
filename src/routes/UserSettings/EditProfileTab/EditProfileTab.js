import PropTypes from 'prop-types'
import React, { Component } from 'react'
import SettingsControl from 'components/SettingsControl'
import SkillsSection from 'components/SkillsSection'
import SkillsToLearnSection from 'components/SkillsToLearnSection'
import Button from 'components/Button'
import Icon from 'components/Icon'
import UploadAttachmentButton from 'components/UploadAttachmentButton'
import Loading from 'components/Loading'
import { bgImageStyle } from 'util/index'
import cx from 'classnames'
import { DEFAULT_BANNER } from 'store/models/Me'

import './EditProfileTab.scss'

const { object, func, string } = PropTypes

const mapSocialProviderToKey = {
  'twitter': 'twitterName',
  'linkedin': 'linkedinUrl',
  'facebook': 'facebookUrl'
}

/** LinkedIn Url */
const validateLinkedinUrl = url => url.match(/^(http(s)?:\/\/)?([\w]+\.)?linkedin\.com/)

export const linkedinPrompt = () => {
  let linkedinUrl = window.prompt('Please enter the full url for your LinkedIn page.')

  while (!validateLinkedinUrl(linkedinUrl)) {
    linkedinUrl = window.prompt('Invalid url. Please enter the full url for your LinkedIn page.')
  }

  return linkedinUrl
}
export class SocialControl extends Component {
  static propTypes = {
    label: string,
    provider: string,
    value: string | null,
    updateSocialSetting: func,
    handleUnlinkAccount: func
  }

  handleLinkClick () {
    const { provider, updateSocialSetting } = this.props

    switch (provider) {
      case 'twitter': {
        const twitterHandle = window.prompt('Please enter your twitter name.')
        if (twitterHandle) {
          updateSocialSetting({ key: mapSocialProviderToKey[provider], value: twitterHandle })
        }
        break
      }
      case 'linkedin': {
        const linkedinUrl = linkedinPrompt()
        updateSocialSetting({ key: mapSocialProviderToKey[provider], value: linkedinUrl })
        break
      }
      // case 'facebook': {
      //   return onLink()
      //   .then(({ error }) => {
      //     if (error) return onChange(false)
      //     return onChange(true)
      //   })
      // }
    }
  }

  handleUnlinkClick () {
    const { handleUnlinkAccount, updateSocialSetting, provider } = this.props

    handleUnlinkAccount()
    updateSocialSetting({ key: mapSocialProviderToKey[provider], value: null })
  }

  render () {
    const { label, value = '' } = this.props
    const linked = !!value

    const linkButton =
      <span
        styleName='link-button'
        onClick={linked ? () => this.handleUnlinkClick() : () => this.handleLinkClick()}>
        {linked ? 'Unlink' : 'Link'}
      </span>

    return (
      <div styleName='control'>
        <div styleName={cx('social-control-label')}>
          {linked ? <Icon name='Complete' styleName='linkedIcon' /> : ''}
          {label}{linkButton}
        </div>
      </div>
    )
  }
}
export default class EditProfileTab extends Component {
  static propTypes = {
    currentUser: object,
    updateUserSettings: func,
    loginWithService: func
  }

  constructor (props) {
    super(props)
    this.state = { edits: {}, changed: false }
  }

  componentDidMount () {
    this.setEditState()
  }

  componentDidUpdate (prevProps, prevState) {
    if (prevProps.fetchPending && !this.props.fetchPending) {
      this.setEditState()
    }
  }

  setEditState () {
    const { currentUser } = this.props
    if (!currentUser) return

    const {
      name, avatarUrl, bannerUrl, tagline, bio,
      contactEmail, contactPhone, locationId, location,
      url, facebookUrl, twitterName, linkedinUrl
    } = currentUser

    this.setState({
      edits: {
        name: name || '',
        avatarUrl: avatarUrl || '',
        bannerUrl: bannerUrl || DEFAULT_BANNER,
        tagline: tagline || '',
        bio: bio || '',
        contactPhone: contactPhone || '',
        contactEmail: contactEmail || '',
        location: location || '',
        locationId: locationId || null,
        url: url || '',
        facebookUrl,
        twitterName,
        linkedinUrl
      }
    })
  }

  updateSetting = (key, setChanged = true) => event => {
    const { edits, changed } = this.state
    setChanged && this.props.setConfirm('You have unsaved changes, are you sure you want to leave?')

    if (key === 'location') {
      edits['location'] = event.target.value.fullText
      edits['locationId'] = event.target.value.id
    } else {
      edits[key] = event.target.value
    }

    this.setState({
      changed: setChanged ? true : changed,
      edits: { ...edits }
    })
  }

  updateSettingDirectly = (key, changed) => value =>
    this.updateSetting(key, changed)({ target: { value } })

  updateSocialSetting = ({ key, value }) => {
    this.setState({
      changed: true,
      edits: { ...this.state.edits, [key]: value }
    })
  }

  save = () => {
    this.setState({ changed: false })
    this.props.setConfirm(false)
    this.props.updateUserSettings(this.state.edits)
  }

  render () {
    const {
      fetchPending,
      currentUser,
      loginWithService,
      unlinkAccount
    } = this.props
    if (fetchPending || !currentUser) return <Loading />

    const { edits, changed } = this.state
    const {
      name, avatarUrl, bannerUrl, tagline, bio,
      contactEmail, contactPhone, location, url,
      facebookUrl, twitterName, linkedinUrl
    } = edits

    const locationObject = currentUser.locationObject

    return <div>
      <input type='text' styleName='name' onChange={this.updateSetting('name')} value={name || ''} />
      <div style={bgImageStyle(bannerUrl)} styleName='banner'>
        <UploadAttachmentButton
          type='userBanner'
          id={currentUser.id}
          onSuccess={({ url }) => this.updateSettingDirectly('bannerUrl')(url)}
          styleName='change-banner-button' />
      </div>
      <UploadAttachmentButton
        type='userAvatar'
        id={currentUser.id}
        onSuccess={({ url }) => this.updateSettingDirectly('avatarUrl')(url)}
        styleName='change-avatar-button'
      >
        <div style={bgImageStyle(avatarUrl)} styleName='avatar'><Icon name='AddImage' styleName='uploadIcon' /></div>
      </UploadAttachmentButton>
      <SettingsControl label='Tagline' onChange={this.updateSetting('tagline')} value={tagline} maxLength={60} />
      <SettingsControl label='About Me' onChange={this.updateSetting('bio')} value={bio} type='textarea' />
      <SettingsControl
        label='Location'
        onChange={this.updateSettingDirectly('location', true)}
        location={location}
        locationObject={locationObject}
        type='location'
      />
      <SettingsControl label='Website' onChange={this.updateSetting('url')} value={url} />
      <SettingsControl label='My Skills &amp; Interests' renderControl={() =>
        <SkillsSection personId={currentUser.id} />} />
      <SettingsControl label='What I&apos;m learning' renderControl={() =>
        <SkillsToLearnSection personId={currentUser.id} />} />
      <SettingsControl label='Contact Email' onChange={this.updateSetting('contactEmail')} value={contactEmail} />
      <SettingsControl label='Contact Phone' onChange={this.updateSetting('contactPhone')} value={contactPhone} />
      <label styleName='social-label'>Social Accounts</label>
      <SocialControl
        label='Facebook'
        provider='facebook'
        value={facebookUrl}
        onLink={() => loginWithService('facebook')}
        updateSocialSetting={this.updateSocialSetting}
        handleUnlinkAccount={() => unlinkAccount('facebook')}
      />
      <SocialControl
        label='Twitter'
        provider='twitter'
        value={twitterName}
        updateSocialSetting={this.updateSocialSetting}
        handleUnlinkAccount={() => unlinkAccount('twitter')}
      />
      <SocialControl
        label='LinkedIn'
        provider='linkedin'
        value={linkedinUrl}
        updateSocialSetting={this.updateSocialSetting}
        handleUnlinkAccount={() => unlinkAccount('linkedin')}
      />
      <div styleName='saveChanges'>
        <span styleName={changed ? 'settingChanged' : ''}>{changed ? 'Changes not saved' : 'Current settings up to date'}</span>
        <Button label='Save Changes' color={changed ? 'green' : 'gray'} onClick={changed ? this.save : null} styleName='save-button' />
      </div>
    </div>
  }
}
