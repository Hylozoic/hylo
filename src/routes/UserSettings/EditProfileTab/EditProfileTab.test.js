import EditProfileTab, { SocialControl, linkedinPrompt } from './EditProfileTab'
import { shallow } from 'enzyme'
import React from 'react'

describe('linkedinPrompt', () => {
  var oldPrompt

  beforeAll(() => {
    oldPrompt = window.prompt
  })

  afterAll(() => {
    window.prompt = oldPrompt
  })

  it('returns a correct linkedIn Url', () => {
    const url = 'https://www.linkedin.com/in/username/'
    window.prompt = jest.fn(() => url)
    expect(linkedinPrompt()).toEqual(url)
  })

  it('rejects an incorrect linkedIn Url, calling itself again', () => {
    var count = 0
    window.prompt = jest.fn(() => {
      if (count === 0) {
        count += 1
        return 'a bad url'
      } else {
        count += 1
        return 'https://www.linkedin.com/in/username/'
      }
    })
    expect(linkedinPrompt()).toEqual('https://www.linkedin.com/in/username/')
    expect(count).toEqual(2)
    expect(window.prompt.mock.calls).toMatchSnapshot()
  })
})

describe('EditProfileTab', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<EditProfileTab currentUser={{}} />)
    expect(wrapper.find('Connect(UploadAttachmentButton)').length).toEqual(2)
    expect(wrapper.find('SettingsControl').length).toEqual(8)
    expect(wrapper.find('SocialControl').length).toEqual(3)
    expect(wrapper.find('Button').prop('color')).toEqual('gray')
    wrapper.setState({ changed: true })
    expect(wrapper.find('Button').prop('color')).toEqual('green')
  })
})

describe('SocialControl', () => {
  it('renders correctly without a value', () => {
    const wrapper = shallow(<SocialControl label='A Social Control' />)
    expect(wrapper.text()).toEqual('A Social ControlLink')
  })

  it('renders correctly with a value', () => {
    const wrapper = shallow(<SocialControl label='A Social Control' value='someurl.com' />)
    expect(wrapper.text()).toEqual('<Icon />A Social ControlUnlink')
  })

  it('calls handleLinkClick when link is clicked', () => {
    const wrapper = shallow(<SocialControl label='A Social Control' />)
    wrapper.instance().handleLinkClick = jest.fn()
    wrapper.find("[data-stylename='link-button']").simulate('click')
    expect(wrapper.instance().handleLinkClick).toHaveBeenCalled()
  })

  it('calls handleUnlinkClick when unlink is clicked', () => {
    const wrapper = shallow(<SocialControl label='A Social Control' value='someurl.com' />)
    wrapper.instance().handleUnlinkClick = jest.fn()
    wrapper.find("[data-stylename='link-button']").simulate('click')
    expect(wrapper.instance().handleUnlinkClick).toHaveBeenCalled()
  })

  describe('handleLinkClick', () => {
    describe('when provider is twitter', () => {
      it('and handle is entered, it updates the twitterName', () => {
        window.prompt = jest.fn(() => 'twitterhandle')
        const updateSocialSetting = jest.fn()
        const handleUnlinkAccount = jest.fn()

        const wrapper = shallow(
          <SocialControl
            label='Twitter'
            provider='twitter'
            value='twitterhandle'
            updateSocialSetting={updateSocialSetting}
            handleUnlinkAccount={handleUnlinkAccount}
          />
        )

        wrapper.instance().handleLinkClick()
        expect(window.prompt).toBeCalledWith('Please enter your twitter name.');
        expect(updateSocialSetting).toHaveBeenCalledWith({ key: 'twitterName', value: 'twitterhandle' })
      })

      it('and handle is NOT entered, it does NOT update the twitterName', () => {
        window.prompt = jest.fn(() => false)
        const updateSocialSetting = jest.fn()
        const handleUnlinkAccount = jest.fn()

        const wrapper = shallow(
          <SocialControl
            label='Twitter'
            provider='twitter'
            value='twitterhandle'
            updateSocialSetting={updateSocialSetting}
            handleUnlinkAccount={handleUnlinkAccount}
          />
        )

        wrapper.instance().handleLinkClick()
        expect(window.prompt).toBeCalledWith('Please enter your twitter name.');
        expect(updateSocialSetting).not.toHaveBeenCalled()
      })
    })

    describe('when provider is linkedin', () => {
      it('and valid linkedinUrl is provided, it updates the linkedinUrl', () => {
        window.prompt = jest.fn(() => 'linkedin.com/test')
        const updateSocialSetting = jest.fn()
        const handleUnlinkAccount = jest.fn()

        const wrapper = shallow(
          <SocialControl
            label='LinkedIn'
            provider='linkedin'
            value='linkedin.com/test'
            updateSocialSetting={updateSocialSetting}
            handleUnlinkAccount={handleUnlinkAccount}
          />
        )

        wrapper.instance().handleLinkClick()
        expect(window.prompt).toBeCalledWith('Please enter the full url for your LinkedIn page.');
        expect(updateSocialSetting).toHaveBeenCalledWith({ key: 'linkedinUrl', value: 'linkedin.com/test' })
      })
    })

    // describe('when provider is facebook', () => {
    //   it('calls the right things when provider is facebook', async () => {
    //     const provider = 'facebook'
    //     const onLink = () => Promise.resolve({ error: false })
    //     const updateUserSettings = jest.fn()
    //     const onChange = jest.fn()
    //     const wrapper = shallow(<SocialControl
    //       label='A Social Control'
    //       provider={provider}
    //       onLink={onLink}
    //       updateUserSettings={updateUserSettings}
    //       onChange={onChange} />)

    //     await wrapper.instance().linkClicked()
    //     expect(updateUserSettings).not.toHaveBeenCalledWith()
    //     expect(onChange).toHaveBeenCalledWith(true)
    //   })
    // })
  })

  describe('handleUnlinkClick', () => {
    const handleUnlinkAccount = jest.fn()
    const updateSocialSetting = jest.fn()

    const wrapper = shallow(
      <SocialControl
        label='LinkedIn'
        provider='linkedin'
        value='linkedin.com/test'
        updateSocialSetting={updateSocialSetting}
        handleUnlinkAccount={handleUnlinkAccount}
      />
    )

    wrapper.instance().handleUnlinkClick()
    expect(handleUnlinkAccount).toHaveBeenCalled()
    expect(updateSocialSetting).toHaveBeenCalledWith({ key: 'linkedinUrl', value: '' })
  })
})
