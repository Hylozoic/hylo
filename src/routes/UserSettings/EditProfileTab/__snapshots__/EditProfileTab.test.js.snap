// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`facebookPrompt rejects an incorrect facebook Url, calling itself again 1`] = `
Array [
  Array [
    "Please enter the full url for your Facebook page.",
  ],
  Array [
    "Invalid url. Please enter the full url for your Facebook page.",
  ],
]
`;

exports[`linkedinPrompt rejects an incorrect linkedIn Url, calling itself again 1`] = `
Array [
  Array [
    "Please enter the full url for your LinkedIn page.",
  ],
  Array [
    "Invalid url. Please enter the full url for your LinkedIn page.",
  ],
]
`;
